packages/components/upload/src/upload-content.vue

```diff
@@ -95,9 +95,9 @@ const upload = async (rawFile: UploadRawFile) => {
     // origin data: Handle data changes after synchronization tasks are executed
     const originData = props.data
     const beforeUploadPromise = props.beforeUpload(rawFile)
-    beforeData = isObject(props.data) ? cloneDeep(props.data) : props.data
+    beforeData = isPlainObject(props.data) ? cloneDeep(props.data) : props.data
     hookResult = await beforeUploadPromise
-    if (isObject(props.data) && isEqual(originData, beforeData)) {
+    if (isPlainObject(props.data) && isEqual(originData, beforeData)) {
       beforeData = cloneDeep(props.data)
     }
```

Comment:
I don't know why it has to clone when `originData` and `beforeData` are equal.

Maybe we should add a deprecated warming when changing data in the `beforeUpload` hook,  and remove these logics in the next minor version?

Reply:
This code is not mine, and I don’t know it, but I have verified that it is useless, but I still think the author should confirm it @GenerQAQ pls.

Reply:
The operation of clone can refer to the case in test.
`FAIL  packages/components/upload/__tests__/upload.test.tsx > <upload /> > functionality > data support receive async function`
`FAIL  packages/components/upload/__tests__/upload.test.tsx > <upload /> > functionality > upload files and save keyList`

Reply:
https://github.com/element-plus/element-plus/pull/12575#issuecomment-1520352432

Reply:
@GenerQAQ so, i can removei it?

Reply:
> @GenerQAQ so, i can removei it?

If unit tests can pass

Reply:
> If unit tests can pass
I didn't try to be compatible with your scene, so keep it
New features only enhance, not break
@HeftyKoo @GenerQAQ

Reply:
> > If unit tests can pass
> 
> I didn't try to be compatible with your scene, so keep it New features only enhance, not break @HeftyKoo @GenerQAQ

OK